kratos/tests/test_container_expression.py

```diff
@@ -852,6 +852,34 @@ def _Read(self, container_expression, variable):
     def _Evaluate(self, container_expression, variable):
         Kratos.Expression.VariableExpressionIO.Write(container_expression, variable)
 
+    def testDomainSizeExpressionIOCondition(self):
+        condition_exp = Kratos.Expression.ConditionExpression(self.model_part)
+        Kratos.Expression.DomainSizeExpressionIO.Read(condition_exp)
+        numpy_condition_exp = condition_exp.Evaluate()
+        for i, condition in enumerate(self.model_part.Conditions):
+            self.assertAlmostEqual(numpy_condition_exp[i], condition.GetGeometry().DomainSize(), 12)
```

Comment:
just curious: why do you need the `12` tolerance? This expression shouldn't do any arithmetic on the domain sizes it's getting from elements/conditions, so I'd expect it to be exactly identical to `condition.GetGeometry().DomainSize()`.

Reply:
The problem is, how the domain size is computed, it has a square root, and squaring which may not produce the same exact values. :/

Reply:
what? But the expression doesn't compute a square or square root, right? That's done internally in the element/condition, am I correct?

If that's the case, the condition/element should **always** return the same number. Floating point operations may not be **exact**, but they must absolutely be **consistent**.

Reply:
My only worry is, when we do the numpy conversion, whether we will lose something. But lets see :)